tests/adapters/test_prologix.py

```diff
@@ -0,0 +1,49 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2021 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import pytest
+import serial
+
+from pymeasure.adapters import PrologixAdapter
+
+prefix="\n".join(["++auto 0", "++eoi 1", "++eos 2"])+"\n"
+
+def make_adapter(**kwargs):
+    return PrologixAdapter(serial.serial_for_url("loop://", **kwargs))
+
+
+@pytest.mark.parametrize("test_input,expected", [("OUTP", prefix + "OUTP\n"),
+                                                 ("POWER 22 dBm", prefix+ "POWER 22 dBm\n")])
+def test_adapter_write(test_input, expected):
+    adapter = make_adapter(timeout=0.2)
+    adapter.write(test_input)
```

Comment:
As long as you are testing non-binary write, wouldn' it be better to use `adapter.ask()` (and rename the test), as we would then also get to test that often-used part of the adapter API (and it's basically what you do, anyway)?
```
returned = adapter.ask(test_input)
assert returned == expected
```
Same for the serial test below.

Reply:
Here, my intent is not to check that an hypothetical command is working by performing an read+write operation.
Indeed my aim is to check that when I send a write, the data I am writing to the device is what I have in the test vector.
As such, the loopback is very nice since it reports me exactly what has been written.
Using the `ask` or a `write`+`read`, would change the bytes read, because, for example, a `PrologixAdapter`, prefix all the read with a write of "`++read eoi`" string. I think writing a fully comprehensive test suite, it is beyond the scope of this PR.

Reply:
I see, agreed. Thanks for the clarification, and for your patience with my many remarks! :-)